inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -102,15 +103,16 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const In
         manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
         manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
         manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
-
-        manager.set_callback(transformations_callback);
-        manager.run_passes(nGraphFunc);
+        manager.register_pass<ngraph::pass::UnrollTensorIterator>();
```

Comment:
In this case you first apply all transformations to TI body and only after then you apply transformations to original function. Is it expected? Why we can not keep current order?

Reply:
All transformations must be applied to the TensorIterator prior to calling UnrollTI due to performance reasons. Much faster to apply transformations to (1xbody) than to (n x body) after unrolling.

Reply:
Yes I understand. I suggest to make this pipeline more clear:
```
 // Apply transformations to function with not unrolled TIs
 ngraph::pass::Manager manager;
 manager.register_pass<ngraph::pass::CommonOptimizations>();
 manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
 manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
 manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
 manager.set_callback(transformations_callback);
 manager.run_passes(nGraphFunc);

 // Apply all transformations to TensorIterator body
 ngraph::pass::Manager ti_manager;
 ti_manager.register_pass<ngraph::pass::ApplyTransformationsToTIBody>(manager);
 // Unroll will be called after all conversions
 ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
 ti_manager.run_passes(nGraphFunc);
```